educator/src/Dscp/DB/SQLite/Util.hs

```diff
@@ -91,16 +92,16 @@ link_
     :: (table ~ RelationT t a b, _)
     => DatabaseEntity be db (TableEntity table)
     -> table (QGenExpr _ _ _)
-    -> Beam.Q Beam.SqliteSelectSyntax db _ ()
+    -> Beam.Q _ db _ ()
 link_ relation (pk1 :-: pk2) = do
     id1 :-: id2 <- all_ relation
     guard_ (id1 ==. pk1)
     guard_ (id2 ==. pk2)
 
 -- | Check whether the query returns any row.
 checkExists
-    :: (MonadIO m)
-    => Beam.Q Beam.SqliteSelectSyntax db (Beam.QNested _) () -> DBT t w m Bool
+    :: (Beam.MonadBeam cmd be hdl m, _)
```

Comment:
Please, don't leave `PartialTypeSignatures` on.

Reply:
I'd like to, but the thing under `Beam.QNested` _is not exported_ by Beam :peka: